lib/index.js

```diff
@@ -39,7 +39,12 @@ function getText(fieldBoost, doc) {
   } else { // "Enhance."
     text = doc;
     for (var i = 0, len = fieldBoost.deepField.length; i < len; i++) {
-      text = text && text[fieldBoost.deepField[i]];
+      if (Array.isArray(text)) {
+        text = text.map(handleNestedObjectArrayItem(fieldBoost, fieldBoost.deepField.slice(i)));
+        break;
```

Comment:
Agreed, you can delete that. Unless you want me to update the PR for it. Tests all run fine with it deleted here.